### PR TITLE
Migrated content from framework-libraries.md to class-library-overview.md

### LINKED ISSUE #1844: Combine existing Framework Libraries doc with migrated equivalent doc
Existing doc: https://docs.microsoft.com/dotnet/articles/standard/framework-libraries
Equivalent doc being migrated will be at: https://docs.microsoft.com/dotnet/articles/standard/class-library-overview

- [ ] Incorporate content from the migrated doc in the existing one
- [ ] Retire the migrated doc

### PR DESCRIPTION
**Pull request by CarinaClaesson**
## Summary

Migrated the relevant parts.

Contributes to #1844 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/standard/class-library-overview.md](https://github.com/dotnet/docs/blob/d7c25a184c4125b11aff59761507583b38ebd5b9/docs/standard/class-library-overview.md) | [.NET class library overview](https://review.learn.microsoft.com/en-us/dotnet/standard/class-library-overview?branch=pr-en-us-35036) |

### CODE REVIEW

**Comment by CarinaClaesson** [comment]
@dotnet-policy-service agree

File: docs/standard/class-library-overview.md [hunk 1]
```diff
@@ -101,6 +101,33 @@ In addition to the base data types, the <xref:System> namespace contains over 10
...
+.NET includes a set of utility APIs that provide functionality for many important tasks.
+
+* <xref:System.Net.Http.HttpClient> - An API for sending HTTP requests and receiving HTTP responses from a resource identified by a URI.
+* <xref:System.Xml.Linq.XDocument> - An API for loading and querying XML documents with LINQ.
+* <xref:System.IO.StreamReader> - An API for reading files.
+* <xref:System.IO.StreamWriter> - An API for writing files.
+
+## App-Model APIs
+
+There are many app-models that can be used with .NET, for example:
```
Comment (gewarren) [inline docs/standard/class-library-overview.md:125]:
```suggestion
There are many app models that can be used with .NET, for example:
```

File: docs/standard/class-library-overview.md [hunk 2]
```diff
@@ -101,6 +101,33 @@ In addition to the base data types, the <xref:System> namespace contains over 10
...
+
+* [ASP.NET](https://dotnet.microsoft.com/apps/aspnet) - A web framework for building Web sites and services. Supported on Windows, Linux, and macOS (depends on ASP.NET version).
```
Comment (gewarren) [inline docs/standard/class-library-overview.md:127]:
```suggestion
* [ASP.NET](https://dotnet.microsoft.com/apps/aspnet) - A web framework for building web sites and services. Supported on Windows, Linux, and macOS (depends on ASP.NET version).
```

File: docs/standard/class-library-overview.md [hunk 3]
```diff
@@ -101,6 +101,33 @@ In addition to the base data types, the <xref:System> namespace contains over 10
...
+* [Xamarin](https://dotnet.microsoft.com/apps/xamarin) - An app platform for building Android and iOS apps with .NET and C#. Supported on Windows and macOS.
```
Comment (gewarren) [inline docs/standard/class-library-overview.md:128]:
```suggestion
* [.NET MAUI](https://dotnet.microsoft.com/apps/maui) - An app platform for building native apps that run on Windows, macOS, iOS, and Android using C#.
```

**Review by gewarren — COMMENTED** [review]
Thanks so much @CarinaClaesson. I left a few suggestions.

**Review by gewarren — APPROVED** [review]

**Pull request merged**